iothub/service/tests/Exceptions/IotHubServiceExceptionTests.cs

```diff
@@ -0,0 +1,154 @@
+        [DataRow(IotHubServiceErrorCode.IotHubUnauthorizedAccess)]
+        [DataRow(IotHubServiceErrorCode.DeviceMaximumQueueDepthExceeded)]
+        [DataRow(IotHubServiceErrorCode.DeviceAlreadyExists)]
+        [DataRow(IotHubServiceErrorCode.ModuleAlreadyExistsOnDevice)]
+        [DataRow(IotHubServiceErrorCode.MessageTooLarge)]
+        [DataRow(IotHubServiceErrorCode.TooManyDevices)]
+        [DataRow(IotHubServiceErrorCode.PreconditionFailed)]
+        public void IotHubServiceException_Ctor_WithNonTransientErrorCode_Ok(IotHubServiceErrorCode errorCode)
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.Accepted; // set for simplicity
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                errorCode);
+
+            // assert
+            exception.StatusCode.Should().Be(statusCode);
+            exception.Message.Should().Be(Message);
+            exception.IsTransient.Should().BeFalse();
+            exception.TrackingId.Should().BeNull();
+            exception.ErrorCode.Should().Be(errorCode);
+        }
+
+        [TestMethod]
+        [DataRow(IotHubServiceErrorCode.DeviceNotOnline)]
+        [DataRow(IotHubServiceErrorCode.ServerError)]
+        [DataRow(IotHubServiceErrorCode.IotHubQuotaExceeded)]
+        [DataRow(IotHubServiceErrorCode.ServiceUnavailable)]
+        [DataRow(IotHubServiceErrorCode.ThrottlingException)]
+        public void IotHubServiceException_Ctor_WithTransidentErrorCode_Ok(IotHubServiceErrorCode errorCode)
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.Accepted;  // set for simplicity
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                errorCode);
+
+            // assert
+            exception.StatusCode.Should().Be(statusCode);
+            exception.Message.Should().Be(Message);
+            exception.IsTransient.Should().BeTrue();
+            exception.TrackingId.Should().BeNull();
+            exception.ErrorCode.Should().Be(errorCode);
+        }
+
+        [TestMethod]
+        public void IotHubServiceException_UnknownErrorCode_IsNotTransient()
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.MovedPermanently; // not HttpStatusCode.RequestTimeout
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                IotHubServiceErrorCode.Unknown);
+
+            // assert
+            exception.StatusCode.Should().Be(statusCode);
+            exception.Message.Should().Be(Message);
+            exception.IsTransient.Should().BeFalse();
+            exception.TrackingId.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceException_UnknownErrorCode_IsTransient()
```

Comment:
```suggestion
        public void IotHubServiceException_RequestTimeoutErrorCode_IsTransient()
```

Reply:
Oh, I thought you were referring to the actual error code 'Unknown', but you mean one we don't track?